adobe/react-spectrum
# fix(#3128): useDialog autoFocus behavior fails in Storybook example for Modal unmounting trigger example

1. Make sure that Dialog autoFocus actually happens.
2. Add workaround for focus management on close within the Storybook example until #2445 is approved.


Closes #3128

## ✅ Pull Request Checklist:

- [x] Included link to corresponding React Spectrum GitHub Issue #3128 .

## 📝 Test Instructions:

1. Open chrome://settings/accessibility and enable "Show a quick highlight on the focused object."
2. Open https://reactspectrum.blob.core.windows.net/reactspectrum/e4919c579a7caf60ee7ea0dc4102f88d4e406cb5/storybook/index.html?path=/story/modal--unmounting-trigger
3. Tab to focus the Open Popover button within the Storybook example.
4. Press Space key to open the Popover
5. Chrome's focus highlight should render around the Popover dialog itself.
6. Tab to focus the Open modal button within the Popover.
7. Press Space key to open the Modal
8. The Popover will close in the background.
9. Chrome's focus highlight should render around the modal Dialog that opens to replace the Popover when the Dialog receives focus.
10. Press Tab to focus, then Space to activate, the "Close"  CTA button, or press Escape key to close the Modal.
12. The modal will close.
11. Focus should be explicitly restored the the Open Popover button. 

## 🧢 Your Project:

Adobe/Accessibility
